src/Helpers/JWKFetcher.php

```diff
@@ -38,39 +45,53 @@ public function __construct(CacheHandler $cache = null, $guzzleOptions = [])
     }
 
     /**
+     * Convert a certificate to PEM.
+     *
+     * @param string $cert Certificate to convert.
      *
-     * @param  string $cert
      * @return string
      */
     protected function convertCertToPem($cert)
     {
-        return '-----BEGIN CERTIFICATE-----'.PHP_EOL.chunk_split($cert, 64, PHP_EOL).'-----END CERTIFICATE-----'.PHP_EOL;
+        $output  = '-----BEGIN CERTIFICATE-----'.PHP_EOL;
+        $output .= chunk_split($cert, 64, PHP_EOL);
+        $output .= '-----END CERTIFICATE-----'.PHP_EOL;
+        return $output;
     }
 
     /**
+     * Fetch
      *
-     * @param string $iss
+     * @param string      $iss       Issuer for the JWKS being fetched.
+     * @param null|string $jwks_path Path to the JWKS from the issuer domain.
      *
      * @return array|mixed|null
-     *
-     * @throws \Exception
      */
-    public function fetchKeys($iss)
+    public function fetchKeys($iss, $jwks_path = null)
     {
-        $url = "{$iss}.well-known/jwks.json";
+        if (empty( $jwks_path )) {
+            $jwks_path = '.well-known/jwks.json';
+        }
 
-        if (($secret = $this->cache->get($url)) === null) {
+        $url = $iss.$jwks_path;
+
+        $secret = $this->cache->get($url);
+        if (is_null($secret)) {
```

Comment:
I'd return early here. If secret is NOT null, return secret. Else .... (avoiding the long indentation, helping readability)

Reply:
Derp ... I should have caught that.